src/d3d10/d3d10core.cpp

```diff
@@ -1,14 +1,41 @@
 
-#include "d3d10.h"
+#include "com/com_object.hpp"
+#include "com/com_pointer.hpp"
+#include "d3d11.h"
 #include "log/log.hpp"
 
 namespace dxmt {
 Logger Logger::s_instance("d3d10core.log");
 
-extern "C" HRESULT STDMETHODCALLTYPE D3D10CoreCreateDevice(
-    IDXGIFactory *pFactory, IDXGIAdapter *pAdapter, UINT Flags,
-    D3D_FEATURE_LEVEL FeatureLevel, ID3D10Device **ppDevice) {
-  ERR("D3D10CoreCreateDevice: stub");
+extern "C" HRESULT WINAPI D3D11CoreCreateDevice(
+    IDXGIFactory *pFactory, IDXGIAdapter *pAdapter, UINT Flags, const D3D_FEATURE_LEVEL *pFeatureLevels,
+    UINT FeatureLevels, ID3D11Device **ppDevice
+);
+
+extern "C" HRESULT STDMETHODCALLTYPE
+D3D10CoreCreateDevice(
+    IDXGIFactory *pFactory, IDXGIAdapter *pAdapter, UINT Flags, D3D_FEATURE_LEVEL FeatureLevel, ID3D10Device **ppDevice
+) {
+  InitReturnPtr(ppDevice);
+
+  Com<ID3D11Device> d3d11_device;
+
+  HRESULT hr = pAdapter->CheckInterfaceSupport(__uuidof(ID3D10Device), nullptr);
+
+  if (FAILED(hr))
+    return hr;
+
+  hr = D3D11CoreCreateDevice(pFactory, pAdapter, Flags | 0x80000000 /* DXMT_D3D10_DEVICE */, &FeatureLevel, 1, &d3d11_device);
```

Comment:
D3D10 defaults to MULTITHREADED so you should probably pass the equivalent D3D11 flag here.0